cranelift/codegen/src/isa/zkasm/inst/emit.rs

```diff
@@ -1366,10 +1368,24 @@ impl MachInstEmit for Inst {
                 put_string(&format!("$ => {} :{opcode}\n", reg_name(rd.to_reg())), sink);
 
                 if cc == IntCC::NotEqual {
-                    put_string("1 => B\n", sink);
+                    let one: u64 = if ty == types::I32 {
+                        1
+                    } else if ty == types::I64 {
+                        4294967296
```

Comment:
Not sure I follow why this needs to be done. If comparison puts a `1` in `A`, then after `XOR` the value in `A` still stays non-zero (i.e. truthy.)

Reply:
afaik it legal to write something like this:
```
i32.const 3
i32.const 5
i32.neq
i32.const 1
i32.add
i32.const 2
call $assert_eq
```

And in general comp operations return always 1 or 0 as i32 and it will be nice to keep this numbers corresponding our i32 representation

Reply:
You are correct of course, but it will still produce the same `1` or `0` in register `A` regardless of whether 32-bit integers or 64-bit integers are being compared. So I’m still not following what this `2^32` is achieving here.